packages/acceptance-tests/pkg-tests-specs/sources/features/plugins.utility.ts

```diff
@@ -0,0 +1,49 @@
+import {PortablePath, xfs} from '@yarnpkg/fslib';
+import https               from 'https';
+import {AddressInfo}       from 'net';
+import {tests}             from 'pkg-tests-core';
+
+export async function createMockPlugin(path: string) {
+  const helloWorldPluginPath = require.resolve(`@yarnpkg/monorepo/scripts/plugin-hello-world.js`) as PortablePath;
+  const helloWorldPlugin = await xfs.readFilePromise(helloWorldPluginPath);
+  const mockPluginPath = `.yarn/plugins/@yarnpkg/plugin-hello-world.cjs`;
+
+  await xfs.mkdirPromise(`${path}/.yarn/plugins/@yarnpkg` as PortablePath, {recursive: true});
+  await xfs.writeFilePromise(`${path}/${mockPluginPath}` as PortablePath, helloWorldPlugin);
+
+  return mockPluginPath as PortablePath;
+}
+
+export async function mockPluginServer(path: PortablePath) {
+  const certs = await tests.getHttpsCertificates();
+  const httpsCaFilePath = `${path}/rootCA.crt` as PortablePath;
+  await xfs.writeFilePromise(httpsCaFilePath, certs.ca.certificate);
+
+  const helloWorldPluginPath = require.resolve(`@yarnpkg/monorepo/scripts/plugin-hello-world.js`) as PortablePath;
+  const helloWorldPlugin = await xfs.readFilePromise(helloWorldPluginPath);
+
+  const server = https.createServer({
+    cert: certs.server.certificate,
+    key: certs.server.clientKey,
+    ca: certs.ca.certificate,
+  }, (req, res) => {
+    res.writeHead(200);
+    res.end(helloWorldPlugin);
+  });
```

Comment:
This function is currently:

```
const {mockServer} = await mockPluginServer();

// ...
```

Can you turn it into this, and add an automatic cleanup when the async callback returns? It's a pattern we use a lot to avoid having to worry about resources leaking (for example, see `mktempPromise`):

```
await mockPluginServer(async mockServer => {
  // ...
});
```

Reply:
If I understand correctly, I'm done 😎